src/app/client/src/app/modules/telemetry/services/telemetry/telemetry.service.ts

```diff
@@ -300,4 +309,18 @@ export class TelemetryService {
       raw: window.navigator.userAgent
     };
   }
+
+  public addUTM(object) {
+    if (sessionStorage.getItem('UTM')) {
```

Comment:
why do we need to get these values from session for every event. Cant we keep these values as a property in telemetry service. create getter and setter for this. Setter can be used to set values for this from app.component also. Why do we need to set value in session in app.component and get value from session in telemetry service.

Reply:
Done with storing inside telemetry service and making 1 copy to session storage also.